src/emulator/controller.ts

```diff
@@ -321,8 +325,11 @@ export async function startAll(options: any, showUI: boolean = true): Promise<vo
     "emulators",
     `Starting emulators: ${targets.join(", ")}`
   );
-  if (options.only) {
-    const requested: string[] = options.only.split(",");
+  const onlyOptions: string = options.only;
```

Comment:
Do we have to repeat this code block twice?

Reply:
Yes. It likely can be refactored in the future, but since it's more lines to refactor than it would be to duplicate, I'm leaving it as-written.